### ISSUE TITLE
Hadolint linux binary v2.4.0 is shipped with a wrong version

### ISSUE THREAD

**Issue by slmg** [hadolint/hadolint#636]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Downloading hadolint `v2.4.0`.

### Actual behavior

Downloaded `v2.3.0-no-git` instead.

### Steps to reproduce the behavior

```sh
> curl -sSL  https://github.com/hadolint/hadolint/releases/download/v2.4.0/hadolint-Linux-x86_64 -o hadolint
> chmod +x hadolint
> ./hadolint --version
Haskell Dockerfile Linter 2.3.0-no-git
```

**Additional environment details (OS, stack version, etc.)**

Debian buster.

**Comment by lorenzo** [comment]
unclear for me to say as the release script has not changed in a while so it is a unidentified bug in our GitHub action workflow.

can anyone help debug what happened?

**Comment by slmg** [comment]
The binaries are different, but that could be due to a dependency change at build time.

```sh
> curl -L https://github.com/hadolint/hadolint/releases/download/v2.4.0/hadolint-Linux-x86_64 -o hadolint-2.4
> curl -L https://github.com/hadolint/hadolint/releases/download/v2.3.0/hadolint-Linux-x86_64 -o hadolint-2.3
> sha256sum hadolint-2.3 
048ca227d596df3d9d0a461d35fe8d082ed4f8f73cf266e77f5d0ee37ca4c78f  hadolint-2.3
> sha256sum hadolint-2.4
aacf288beb8422dbdca79adf29f4e2be7afc227c68bf4984a39914a3aede1234  hadolint-2.4
```

**Comment by m-ildefons** [comment]
I believe this is caused by committing a `package.yaml` containing 2.4.0 together with a `hadolint.cabal` containing 2.3.0. since the release process only calls cabal and not stack, the cabal file does not get regenerated thus baking in the older version number:
https://github.com/hadolint/hadolint/blob/7c505591f2b254023e862a65d557fa62f6cccc18/.github/workflows/release.yml#L87

**Comment by darkvertex** [comment]
Maybe since releases are meant to be immutable and there was a minor typo fix and the codebase did change, it should be bumped to v2.4.1? and then everything will be good.

FYI @lorenzo v2.4.0 in the Releases page of the repo as of the time I'm typing this, is still the old build.

**Comment by lorenzo** [comment]
The build is new, only the version number in the binary is wrong. I’ll publish a new version soon when I get a few minutes
